src/eureka/S3_data_reduction/optspex.py

```diff
@@ -477,9 +478,99 @@ def profile_gauss(subdata, mask, threshold=10, guess=None, isplots=0):
     return profile
 
 
-def optimize(meta, subdata, mask, bg, spectrum, Q, v0, p5thresh=10,
-             p7thresh=10, fittype='smooth', window_len=21, deg=3,
-             windowtype='hanning', n=0, m=0, meddata=None):
+def optimize(data, meta, log, apdata, apmask, apbg, apv0, gain=1,
```

Comment:
I would call this routine 'optimize_wrapper' and leave optimize as previously defined.